src/main/java/javaslang/collection/List.java

```diff
@@ -278,6 +278,40 @@
         return List.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Returns a List containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the List
+     * @param n The number of elements in the List
+     * @param f The Function computing element values
+     * @return A List consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code f} is null
+     */
+    static <T> List<T> tabulate(int n, Function<? super Integer, ? extends T> f) {
+        Objects.requireNonNull(f, "f is null");
+        int nOrZero = java.lang.Math.max(n, 0);
+        @SuppressWarnings("unchecked")
+        T[] elements = (T[]) new Object[nOrZero];
+        for (int i = 0; i < n; i++) {
+            elements[i] = f.apply(i);
+        }
+        return List.of(elements);
```

Comment:
I avoided this on purpose, my point is that doing in this way we call f in the reverse order, from f(n-1) to f(0). In case of pure functions this is fine, but if f is a weird stateful function, we get some inconsistencies between different implementations of Seq.
For example, given the following function
```
java.util.LinkedList<Integer> ints = new java.util.LinkedList<>(Arrays.asList(0, 1, 2));
Function<Integer, Integer> f = i -> ints.remove();
```
Array.fill(3, f) returns Array(0, 1, 2) but List.fill(3, f) returns List(2, 1, 0)